src/Mod/Draft/draftutils/gui_utils.py

```diff
@@ -98,50 +99,68 @@ def autogroup(obj):
-        active_arch_obj = FreeCADGui.ActiveDocument.ActiveView.getActiveObject("Arch")
-        if hasattr(FreeCADGui,"draftToolBar"):
-            if (hasattr(FreeCADGui.draftToolBar,"autogroup") 
-                and not FreeCADGui.draftToolBar.isConstructionMode()
-                ):
-                if FreeCADGui.draftToolBar.autogroup is not None:
-                    active_group = FreeCAD.ActiveDocument.getObject(FreeCADGui.draftToolBar.autogroup)
-                    if active_group:
-                        found = False
-                        for o in active_group.Group:
-                            if o.Name == obj.Name:
-                                found = True
-                        if not found:
-                            gr = active_group.Group
-                            gr.append(obj)
-                            active_group.Group = gr
-                elif active_arch_obj:
-                    active_arch_obj.addObject(obj)
-                elif FreeCADGui.ActiveDocument.ActiveView.getActiveObject("part", False) is not None:
-                    # add object to active part and change it's placement accordingly
-                    # so object does not jump to different position, works with App::Link
-                    # if not scaled. Modified accordingly to realthunder suggestions
-                    p, parent, sub = FreeCADGui.ActiveDocument.ActiveView.getActiveObject("part", False)
-                    matrix = parent.getSubObject(sub, retType=4)
-                    if matrix.hasScale() == 1:
-                        FreeCAD.Console.PrintMessage(translate("Draft",
-                                                               "Unable to insert new object into "
-                                                               "a scaled part")
-                                                    )
-                        return
-                    inverse_placement = FreeCAD.Placement(matrix.inverse())
-                    if get_type(obj) == 'Point':
-                        # point vector have a kind of placement, so should be
-                        # processed before generic object with placement
-                        point_vector = FreeCAD.Vector(obj.X, obj.Y, obj.Z)
-                        real_point = inverse_placement.multVec(point_vector)
-                        obj.X = real_point.x
-                        obj.Y = real_point.y
-                        obj.Z = real_point.z
-                    elif hasattr(obj,"Placement"):
-                        obj.Placement = FreeCAD.Placement(inverse_placement.multiply(obj.Placement))
-                    p.addObject(obj)
+    
+    # check for required conditions for autogroup to work
+    if not App.GuiUp:
+        return
+    if not hasattr(Gui,"draftToolBar"):
+        return
+    if not hasattr(Gui.draftToolBar,"autogroup"):
+        return        
+    if Gui.draftToolBar.isConstructionMode():
+        return
```

Comment:
Well done here. We definite can optimize this in many Gui commands. Instead of using very nested ifs, in many cases we can return immediately if the conditions are not set. I haven't done much about this, but it's something we should eventually change in many places.

By the way, I'd try to change the number of ifs in some cases, as long as the line is less than 80 characters.
```py
if not hasattr(Gui, "draftToolBar") or not hasattr(Gui.draftToolBar, "autogroup"):
    return
```

In this case, it returns in the first test, and doesn't even run the second one.